DiskThresholdDeciderIT failing  in 7.x

**Build scan**:

https://gradle-enterprise.elastic.co/s/werdnzcryjnus

**Repro line**:

```
REPRODUCE WITH: gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded" \
  -Dtests.seed=DA54BEE5EB921E46 \
  -Dtests.security.manager=true \
  -Dtests.locale=tr-TR \
  -Dtests.timezone=SystemV/AST4ADT \
  -Druntime.java=8

REPRODUCE WITH: gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testRestoreSnapshotAllocationDoesNotExceedWatermark" \
  -Dtests.seed=DA54BEE5EB921E46 \
  -Dtests.security.manager=true \
  -Dtests.locale=tr-TR \
  -Dtests.timezone=SystemV/AST4ADT \
  -Druntime.java=8

REPRODUCE WITH: gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded" \
  -Dtests.seed=DA54BEE5EB921E46 \
  -Dtests.security.manager=true \
  -Dtests.locale=tr-TR \
  -Dtests.timezone=SystemV/AST4ADT \
  -Druntime.java=8

REPRODUCE WITH: gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testRestoreSnapshotAllocationDoesNotExceedWatermark" \
  -Dtests.seed=DA54BEE5EB921E46 \
  -Dtests.security.manager=true \
  -Dtests.locale=tr-TR \
  -Dtests.timezone=SystemV/AST4ADT \
  -Druntime.java=8
```

**Reproduces locally?**:

Yes

**Applicable branches**:

7.x

**Failure history**:


**Failure excerpt**:

```
12:37:02 org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT > testHighWatermarkNotExceeded FAILED
12:37:02     java.lang.AssertionError: C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2016\server\build\testrun\internalClusterTest\temp\org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT_DA54BEE5EB921E46-001\tempDir-003\node-0\nodes\0 not tracked and not on Linux
12:37:02         at __randomizedtesting.SeedInfo.seed([DA54BEE5EB921E46:33755F576B54D7A8]:0)
12:37:02         at org.junit.Assert.fail(Assert.java:88)
12:37:02         at org.junit.Assert.assertTrue(Assert.java:41)
12:37:02         at org.elasticsearch.cluster.DiskUsageIntegTestCase$TestFileSystemProvider.getTestFileStore(DiskUsageIntegTestCase.java:222)
12:37:02         at org.elasticsearch.cluster.DiskUsageIntegTestCase$TestFileSystemProvider.getFileStore(DiskUsageIntegTestCase.java:211)
12:37:02         at java.nio.file.Files.getFileStore(Files.java:1461)
12:37:02         at org.elasticsearch.env.Environment.getFileStore(Environment.java:328)
12:37:02         at org.elasticsearch.env.NodeEnvironment$NodePath.<init>(NodeEnvironment.java:113)
12:37:02         at org.elasticsearch.env.NodeEnvironment$NodeLock.<init>(NodeEnvironment.java:218)
12:37:02         at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:272)
12:37:02         at org.elasticsearch.node.Node.<init>(Node.java:362)
12:37:02         at org.elasticsearch.node.MockNode.<init>(MockNode.java:99)
12:37:02         at org.elasticsearch.node.MockNode.<init>(MockNode.java:89)
12:37:02         at org.elasticsearch.test.InternalTestCluster.buildNode(InternalTestCluster.java:737)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startNodes(InternalTestCluster.java:2221)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startNode(InternalTestCluster.java:2160)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startMasterOnlyNode(InternalTestCluster.java:2295)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startMasterOnlyNode(InternalTestCluster.java:2287)
12:37:02         at org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded(DiskThresholdDeciderIT.java:79)
12:37:02 REPRODUCE WITH: gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded" -Dtests.seed=DA54BEE5EB921E46 -Dtests.security.manager=true -Dtests.locale=tr-TR -Dtests.timezone=SystemV/AST4ADT -Druntime.java=8
12:37:02 
12:37:02 org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT > testRestoreSnapshotAllocationDoesNotExceedWatermark FAILED
12:37:02     java.lang.AssertionError: C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2016\server\build\testrun\internalClusterTest\temp\org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT_DA54BEE5EB921E46-001\tempDir-005\node-0\nodes\0 not tracked and not on Linux
12:37:02         at __randomizedtesting.SeedInfo.seed([DA54BEE5EB921E46:C8EC1884802B01FE]:0)
12:37:02         at org.junit.Assert.fail(Assert.java:88)
12:37:02         at org.junit.Assert.assertTrue(Assert.java:41)
12:37:02         at org.elasticsearch.cluster.DiskUsageIntegTestCase$TestFileSystemProvider.getTestFileStore(DiskUsageIntegTestCase.java:222)
12:37:02         at org.elasticsearch.cluster.DiskUsageIntegTestCase$TestFileSystemProvider.getFileStore(DiskUsageIntegTestCase.java:211)
12:37:02         at java.nio.file.Files.getFileStore(Files.java:1461)
12:37:02         at org.elasticsearch.env.Environment.getFileStore(Environment.java:328)
12:37:02         at org.elasticsearch.env.NodeEnvironment$NodePath.<init>(NodeEnvironment.java:113)
12:37:02         at org.elasticsearch.env.NodeEnvironment$NodeLock.<init>(NodeEnvironment.java:218)
12:37:02         at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:272)
12:37:02         at org.elasticsearch.node.Node.<init>(Node.java:362)
12:37:02         at org.elasticsearch.node.MockNode.<init>(MockNode.java:99)
12:37:02         at org.elasticsearch.node.MockNode.<init>(MockNode.java:89)
12:37:02         at org.elasticsearch.test.InternalTestCluster.buildNode(InternalTestCluster.java:737)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startNodes(InternalTestCluster.java:2221)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startNode(InternalTestCluster.java:2160)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startMasterOnlyNode(InternalTestCluster.java:2295)
12:37:02         at org.elasticsearch.test.InternalTestCluster.startMasterOnlyNode(InternalTestCluster.java:2287)
12:37:02         at org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testRestoreSnapshotAllocationDoesNotExceedWatermark(DiskThresholdDeciderIT.java:109)
```

--- a/test/framework/src/main/java/org/elasticsearch/cluster/DiskUsageIntegTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/cluster/DiskUsageIntegTestCase.java
@@ -212,6 +212,9 @@ public class DiskUsageIntegTestCase extends ESIntegTestCase {
         }
 
         TestFileStore getTestFileStore(Path path) {
+            if (path.endsWith(path.getFileSystem().getPath("nodes", "0"))) {
+                path = path.getParent().getParent();
+            }
             final TestFileStore fileStore = trackedPaths.get(path);
             if (fileStore != null) {
                 return fileStore;

